Subject: 0001-upstream-avoid-download-to-server-controlled-path-wh.patch: upstream: avoid download to server-controlled path when
CVE backport for Alpine 3.23 openssh 10.2_p1-r1 (aports 3.24-stable patches applied onto the tracked 10.2p1 tree).
Tracker: https://nvd.nist.gov/vuln/detail/CVE-2026-59995
Upstream source: https://github.com/openssh/openssh-portable/commit/1b39f39657d2e58f8ec57341581a39bbf0be645b
Alpine aports patch: https://github.com/alpinelinux/aports/blob/da03ee144d27cece40d2a293c83214592415ae91/main/openssh/0001-upstream-avoid-download-to-server-controlled-path-wh.patch
Base tree: OpenSSH portable 10.2p1 (https://ftp.openbsd.org/pub/OpenBSD/OpenSSH/portable/openssh-10.2p1.tar.gz, sha256:ccc42c0419937959263fa1dbd16dafc18c56b984c03562d2937ce56a60f798b2), extracted as openssh-10.2p1/
Files: sftp.c

Ported verbatim from Alpine aports 3.24-stable commit da03ee144d27cece40d2a293c83214592415ae91; the diff applies unchanged onto OpenSSH portable 10.2p1.

From 837b4e526fd9817707549d68af6cd6db6e7903f6 Mon Sep 17 00:00:00 2001
From: "djm@openbsd.org" <djm@openbsd.org>
Date: Mon, 29 Jun 2026 01:47:21 +0000
Subject: [PATCH 01/12] upstream: avoid download to server-controlled path when
 performing

download on the commandline. From Swival scanner

OpenBSD-Commit-ID: d1b2c44305fdfe6d51eed9ecc727e59478bf311f
(cherry picked from commit 1b39f39657d2e58f8ec57341581a39bbf0be645b)
[CVE-2026-59995]
---
 sftp.c | 9 ++-------
 1 file changed, 2 insertions(+), 7 deletions(-)

diff --git a/sftp.c b/sftp.c
index eebb166e8de4..33c8364e35bb 100644
--- a/sftp.c
+++ b/sftp.c
@@ -2287,13 +2287,8 @@ interactive_loop(struct sftp_conn *conn, char *file1, char *file2)
 				return (-1);
 			}
 		} else {
-			/* XXX this is wrong wrt quoting */
-			snprintf(cmd, sizeof cmd, "get%s %s%s%s",
-			    global_aflag ? " -a" : "", dir,
-			    file2 == NULL ? "" : " ",
-			    file2 == NULL ? "" : file2);
-			err = parse_dispatch_command(conn, cmd,
-			    &remote_path, startdir, 1, 0);
+			err = process_get(conn, dir, file2, remote_path, 0, 0,
+			    global_aflag, 0);
 			free(dir);
 			free(startdir);
 			free(remote_path);
-- 
2.55.0.dirty
